drivers/ethernet/eth_e1000.c

```diff
@@ -442,22 +453,29 @@ static const struct ptp_clock_driver_api api = {
 	.rate_adjust = ptp_clock_e1000_rate_adjust,
 };
 
-static int ptp_e1000_init(const struct device *port)
-{
-	const struct device *const eth_dev = DEVICE_DT_INST_GET(0);
-	struct e1000_dev *context = eth_dev->data;
-	struct ptp_context *ptp_context = port->data;
-
-	context->ptp_clock = port;
-	ptp_context->eth_context = context;
-
-	ptp_context->clock_time = k_ticks_to_ns_floor64(k_uptime_ticks());
-
-	return 0;
-}
+#define E1000_PTP_INIT(inst)						\
+	static struct ptp_context ptp_e1000_context_##inst;		\
+	static int ptp_e1000_init_##inst(const struct device *port)	\
+	{								\
+		const struct device *const eth_dev =			\
+					DEVICE_DT_INST_GET(inst);	\
+		struct e1000_dev *context = eth_dev->data;		\
+		struct ptp_context *ptp_context = port->data;		\
+									\
+		context->ptp_clock = port;				\
+		ptp_context->eth_context = context;			\
+									\
+		ptp_context->clock_time =				\
+				k_ticks_to_ns_floor64(k_uptime_ticks());\
+									\
+		return 0;						\
+	}								\
```

Comment:
This certainly works, I just wonder if we could do here something like this because all the instance init functions are basically the same (but with different parameters):
```
static int ptp_e1000_init(const struct device *port, const struct device *const eth_dev)
{
	struct e1000_dev *context = eth_dev->data;
	struct ptp_context *ptp_context = port->data;

	context->ptp_clock = port;
	ptp_context->eth_context = context;

	ptp_context->clock_time = k_ticks_to_ns_floor64(k_uptime_ticks());

	return 0;
}

#define E1000_PTP_INIT(inst)						\
	static struct ptp_context ptp_e1000_context_##inst;		\
	static int ptp_e1000_init_##inst(const struct device *port)	\
	{								\
		const struct device *const eth_dev =			\
					DEVICE_DT_INST_GET(inst);	\
									\
		return ptp_e1000_init(port, eth_dev);		\
	}
```
This would minimize the amount of code that we are duplicating.

Reply:
I made it simplier with

```
static int ptp_e1000_init(const struct device *port)
{
	struct ptp_context *ptp_context = port->data;
	struct e1000_dev *context = ptp_context->eth_context;

	context->ptp_clock = port;
	ptp_context->clock_time = k_ticks_to_ns_floor64(k_uptime_ticks());

	return 0;
}

#define E1000_PTP_INIT(inst)						\
	static struct ptp_context ptp_e1000_context_##inst = {		\
		.eth_context = DEVICE_DT_INST_GET(inst)->data,		\
	};								\
```